### PR TITLE
AMLS-3468 | MDTP Only: New FX sub-sector to MSB Activities (MSB)

### PR DESCRIPTION
**Pull request by drewmiley**

### CODE REVIEW

File: app/controllers/msb/IdentifyLinkedTransactionsController.scala [hunk 1]
```diff
@@ -50,6 +50,8 @@ class IdentifyLinkedTransactionsController @Inject() (val dataCacheConnector: Da
         Redirect(routes.BusinessUseAnIPSPController.get())
       case s if s contains CurrencyExchange =>
         Redirect(routes.CETransactionsInNext12MonthsController.get())
+      case s if s contains ForeignExchange =>
```
Comment (drewmiley) [inline app/controllers/msb/IdentifyLinkedTransactionsController.scala]: Look at routing refactor

File: app/controllers/msb/MostTransactionsController.scala [hunk 2]
```diff
@@ -95,12 +94,16 @@ class MostTransactionsController @Inject()(val authConnector: AuthConnector = AM
       case true =>
         if (services contains CurrencyExchange) {
           Redirect(routes.CETransactionsInNext12MonthsController.get())
+        } else if (services contains ForeignExchange) {
```
Comment (drewmiley) [inline app/controllers/msb/MostTransactionsController.scala]: Look at routing refactor

File: app/controllers/msb/MostTransactionsController.scala [hunk 3]
```diff
@@ -109,6 +112,8 @@ class MostTransactionsController @Inject()(val authConnector: AuthConnector = AM
   private def editRouting(services: Set[BusinessMatchingMsbService], msb: MoneyServiceBusiness) =
     if ((services contains CurrencyExchange) && msb.ceTransactionsInNext12Months.isEmpty) {
       Redirect(routes.CETransactionsInNext12MonthsController.get(true))
+    } else if ((services contains ForeignExchange) && msb.ceTransactionsInNext12Months.isEmpty) {
```
Comment (drewmiley) [inline app/controllers/msb/MostTransactionsController.scala]: Look at routing refactor

File: app/controllers/msb/SendMoneyToOtherCountryController.scala [hunk 4]
```diff
@@ -91,6 +91,10 @@ class SendMoneyToOtherCountryController @Inject()(val dataCacheConnector: DataCa
           Redirect(routes.CETransactionsInNext12MonthsController.get())
         case (false, s, true) if s contains CurrencyExchange =>
           Redirect(routes.CETransactionsInNext12MonthsController.get())
+        case (false, _, false) if shouldAnswerForeignExchangeQuestions(services, register) =>
```
Comment (drewmiley) [inline app/controllers/msb/SendMoneyToOtherCountryController.scala]: Look at routing refactor

File: app/controllers/msb/SendMoneyToOtherCountryController.scala [hunk 5]
```diff
@@ -104,6 +108,9 @@ class SendMoneyToOtherCountryController @Inject()(val dataCacheConnector: DataCa
       case (false, s)
         if (s contains CurrencyExchange) && msb.sendTheLargestAmountsOfMoney.isEmpty =>
         Redirect(routes.CETransactionsInNext12MonthsController.get(true))
+      case (false, s)
```
Comment (drewmiley) [inline app/controllers/msb/SendMoneyToOtherCountryController.scala]: Look at routing refactor

File: app/controllers/msb/WhichCurrenciesController.scala [hunk 6]
```diff
@@ -59,13 +59,33 @@ class WhichCurrenciesController @Inject() (val authConnector: AuthConnector,
...
+              val result = for {
+                cache <- optMap
+                msb <- cache.getEntry[MoneyServiceBusiness](MoneyServiceBusiness.key)
+                bm <- cache.getEntry[BusinessMatching](BusinessMatching.key)
+                services <- bm.msbServices
+              } yield {
+                dataCacheConnector.save[MoneyServiceBusiness](MoneyServiceBusiness.key,
+                  msb.whichCurrencies(data)
+                ) map { _ =>
+                  edit match {
```
Comment (drewmiley) [inline app/controllers/msb/WhichCurrenciesController.scala]: Look at routing refactor

File: conf/application.conf [hunk 7]
```diff
@@ -112,7 +112,7 @@ microservice {
       has-accepted = true
       show-fees = false
       enrolment-store = true
-      fx-enabled = false
+      fx-enabled = true
```
Comment (drewmiley) [inline conf/application.conf]: Should this be false for now?

File: app/controllers/msb/IdentifyLinkedTransactionsController.scala (hunk 1 above)
Comment (drewmiley) [reply app/controllers/msb/IdentifyLinkedTransactionsController.scala]: Some edi case work required

File: test/views/msb/summarySpec.scala [hunk 8]
```diff
@@ -18,18 +18,17 @@ package views.msb
```
Comment (drewmiley) [inline test/views/msb/summarySpec.scala]: More work required on this

File: app/controllers/msb/MostTransactionsController.scala [hunk 9]
```diff
@@ -89,27 +84,38 @@ class MostTransactionsController @Inject()(val authConnector: AuthConnector = AM
...
-    }
+  private def shouldAnswerCurrencyExchangeQuestions(
+                                                           msbServices: Set[BusinessMatchingMsbService],
+                                                           register: ServiceChangeRegister,
+                                                           isPreSubmission: Boolean,
+                                                           msb: MoneyServiceBusiness,
+                                                           edit: Boolean
+                                                   ): Boolean = {
+    currencyExchangeAddedPostSubmission(msbServices, register) ||
+            (isPreSubmission && msbServices.contains(CurrencyExchange) && (msb.ceTransactionsInNext12Months.isEmpty || !edit))
```
Comment (JGRobinson) [inline app/controllers/msb/MostTransactionsController.scala]: Not sure having edit in this method is appropriate

File: app/controllers/msb/MostTransactionsController.scala [hunk 10]
```diff
@@ -89,27 +84,38 @@ class MostTransactionsController @Inject()(val authConnector: AuthConnector = AM
...
-      Redirect(routes.SummaryController.get())
+  private def shouldAnswerForeignExchangeQuestions(
+                                                          msbServices: Set[BusinessMatchingMsbService],
+                                                          register: ServiceChangeRegister,
+                                                          isPreSubmission: Boolean,
+                                                          msb: MoneyServiceBusiness,
+                                                          edit: Boolean
+                                                  ): Boolean = {
+    foreignExchangeAddedPostSubmission(msbServices, register) ||
+            (isPreSubmission && msbServices.contains(ForeignExchange) && (msb.fxTransactionsInNext12Months.isEmpty || !edit))
```
Comment (JGRobinson) [inline app/controllers/msb/MostTransactionsController.scala]: Not sure having edit in this method is appropriate

**Pull request merged**